addons/stock/models/stock_picking.py

```diff
@@ -570,11 +570,23 @@ def create(self, vals):
                         move[2]['company_id'] = picking_type.company_id.id
         res = super(Picking, self).create(vals)
         res._autoconfirm_picking()
+
+        # set partner as follower
+        if vals.get('partner_id'):
+            for picking in res:
+                picking.message_subscribe([vals.get('partner_id')])
+
         return res
 
     def write(self, vals):
         if vals.get('picking_type_id') and self.state != 'draft':
             raise UserError(_("Changing the operation type of this record is forbidden at this point."))
+        # set partner as a follower and unfollow old partner
+        if vals.get('partner_id'):
+            for picking in self:
```

Comment:
``` python3
if not picking.location_dest_id.usage == 'customer':
    continue
```

Reply:
Oma said partner should be add as a follower for both delivery and receipt, so I think we should check `picking.location_id.usage == 'supplier' or picking.location_dest_id.usage == 'customer' `. right?

Reply:
@yhu-odoo :+1: